ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py

```diff
@@ -24,38 +26,92 @@
 import ai_edge_torch.generative.examples.stable_diffusion.diffusion as diffusion
 from ai_edge_torch.generative.examples.stable_diffusion.encoder import Encoder
 import ai_edge_torch.generative.examples.stable_diffusion.util as util
-import ai_edge_torch.generative.utilities.loader as loading_utils
 import ai_edge_torch.generative.utilities.stable_diffusion_loader as stable_diffusion_loader
 
+arg_parser = argparse.ArgumentParser()
+arg_parser.add_argument(
+    '--clip_ckpt', type=str, help='Path to source CLIP model checkpoint', required=True
+)
+arg_parser.add_argument(
+    '--encoder_ckpt',
+    type=str,
+    help='Path to source image encoder model checkpoint',
+    required=False,
+)
+arg_parser.add_argument(
+    '--diffusion_ckpt',
+    type=str,
+    help='Path to source diffusion model checkpoint',
+    required=True,
+)
+arg_parser.add_argument(
+    '--decoder_ckpt',
+    type=str,
+    help='Path to source image decoder model checkpoint',
+    required=True,
+)
+arg_parser.add_argument(
+    '--ckpt_format',
+    type=str,
+    choices=['pytorch', 'safetensors'],
+    default='safetensors',
+    nargs='?',
+    const='safetensors',
+)
+arg_parser.add_argument(
+    '--output_dir',
+    type=str,
+    help='Path to the converted TFLite directory.',
+    required=True,
+)
+
 
 @torch.inference_mode
 def convert_stable_diffusion_to_tflite(
+    ckpt_format: str,
+    output_dir: str,
     clip_ckpt_path: str,
-    encoder_ckpt_path: str,
     diffusion_ckpt_path: str,
     decoder_ckpt_path: str,
+    encoder_ckpt_path: Optional[str] = None,
     image_height: int = 512,
     image_width: int = 512,
 ):
 
-  clip_model = clip.CLIP(clip.get_model_config())
-  loader = loading_utils.ModelLoader(clip_ckpt_path, clip.TENSOR_NAMES)
-  loader.load(clip_model, strict=False)
+  strict_loading = ckpt_format == 'pytorch'
 
-  encoder = Encoder()
-  encoder.load_state_dict(torch.load(encoder_ckpt_path))
+  clip_model = clip.CLIP(
+      clip.get_model_config(qkv_fused_interleaved=(ckpt_format == 'pytorch'))
+  )
```

Comment:
We can potentially simplify the code here if we just support the standard HF checkpoint. The pytorch checkpoint is from a much unpopular repo, and having that support here may not be very useful in the longer term..